docs/reference/commandline/network_connect.md

```diff
@@ -16,14 +16,25 @@ parent = "smn_cli"
 
       --help=false       Print usage
 
-Connects a running container to a network. This enables instant communication with other containers belonging to the same network.
+Connects a running container to a network. Once connected, the container can
+communicate with other containers in the same network.
 
+```bash
+$ docker network connect multi-host-network container1
 ```
-  $ docker network create -d overlay multi-host-network
-  $ docker run -d --name=container1 busybox top
-  $ docker network connect multi-host-network container1
-```
 
-the container will be connected to the network that is created and managed by the driver (multi-host overlay driver in the above example) or external network plugins.
+You can also use the `docker run --net` option to start a container and
+immediately connect it to a network. You can pause, start, and stop containers that are connected to a network. The container retains its connection information. When the container is stopped, it does not appear in the network until you restart it.
+
+To verify the container is connected, use the `docker network inspect` command. Use `docker network disconnect` to remove a container from the network.
+
+You can connect multiple containers to the same network. Once connected, the containers can communicate without the need to expose and publish ports among containers. For `overlay` or custom plugins that support multi-host connectivity, then the containers connected to the same multi-host network will be able to communicate seamlessly.
```

Comment:
And also, may be worth explicitly calling out that containers can talk between Docker Engines if they are attached to the same multi-host capable network. e.g
`For networks that use the `overlay` or custom drivers that support multi-host connectivity, then the containers running on different Docker Engines will be able to communicate seamlessly.`